Redesign

igor-kamil opened the pull request:
# Description

### Changed
- carousel & hero image refactoring, added img-srcset
- unified main image storing for slides, collections and articles
- new styling for artwork / article / collection, metadata blocks

⚠️ to compile assets, it requires to add `SourceSerifPro` and updated version of `GT-Walsheim` 

Fixes https://jira.sng.sk/browse/WEBUMENIA-1547

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

eronisko commented on resources/views/dielo.blade.php:
```diff
@@ -225,7 +175,8 @@ class="btn btn-default btn-xs btn-outline">{!! $tag !!}</a>
                                 <td>
                                     @foreach ($item->techniques as $technique)
-                                    <a href="{!! URL::to('katalog?technique=' . $technique) !!}">{!! $technique !!}</a><br>
+                                    <a href="{!! URL::to('katalog?technique=' . $technique) !!}">{!! $technique
+                                        !!}</a><br>
```
These line breaks look a bit strange (also `{!!` is probably not necessary, but I wouldn't fix that now)
```suggestion
                                    <a href="{!! URL::to('katalog?technique=' . $technique) !!}">{!! $technique !!}</a><br>
```

eronisko commented on resources/views/dielo.blade.php:
```diff
@@ -478,7 +496,8 @@ class="btn btn-default btn-outline uppercase sans">{{ trans('general.close') }}<
 
 @if (!empty($item->lat) && ($item->lat > 0))
-<script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyCVG26BxGY9yhjCFbviWRgZsvxSlikOnIM&callback=initMap" async defer></script>
+<script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyCVG26BxGY9yhjCFbviWRgZsvxSlikOnIM&callback=initMap" async
+        defer></script>
```
```suggestion
<script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyCVG26BxGY9yhjCFbviWRgZsvxSlikOnIM&callback=initMap" async defer></script>
```

eronisko commented on resources/lang/sk/general.php:
```diff
@@ -42,5 +42,7 @@
     "share_link"         => "Zdieľať URL adresu",
     "copy"               => "kopírovať",
     "copied_to_clipboard"=> "Odkaz bol skopírovaný do schránky",
+    'minute'             => ':count minúta|:count minúty|:count minút',
+    'reading_time'       => 'Čas čítania',
```
We probably forgot CS and EN translations for this

eronisko reviewed (changes requested):
Code looks good overall, just minor suggestions in the comments. However, there are some issues that I've encountered.

On public side:
1. In catalogue, clicking on sorting select throws `Uncaught Error: Syntax error, unrecognized expression: #` (but it seems to work nevertheless)

In Admin:
1. Clicking on 'Carousel' (http://127.0.0.1:8000/slide) throws an error: `Undefined class constant 'ARTWORKS_DIR' (View: C:\Users\eronisko\workspace\sng\webumenia\resources\views\slides\index.blade.php)`

1. http://127.0.0.1:8000/article/create generates a bunch of `GET http://127.0.0.1:8000/css/content-slick.css net::ERR_ABORTED 404 (Not Found)`

1. ~~Item editing (e.g. http://127.0.0.1:8000/item/SVK:TMP.1/edit) looks broken -- see JS console~~ Fixed in #458 

1. Every CKEditor form now uses the new (bigger) article font. That's probably not desired.

igor-kamil commented on app/Helpers/general.php:
```diff
@@ -1,5 +1,8 @@
 <?php
 
+use Carbon\Carbon;
+
+
```
it's used in the method `cedvuDatetime($date)` within the helpers (which is btw not being used anymore...)

igor-kamil commented on package.json:
```diff
@@ -12,7 +12,7 @@
     "devDependencies": {
         "cross-env": "^5.1",
-        "jquery": "^3.4.1",
+        "jquery": "^3.5.0",
```
I was also surprised. I know and I remember that. However it doesn't throw any errors anymore... ✨ magic✨

eronisko commented:
@igor-kamil 

> 3. Item editing (e.g. http://127.0.0.1:8000/item/SVK:TMP.1/edit) looks broken -- see JS console 
This is now fixed in #458

igor-kamil commented:
> Code looks good overall, just minor suggestions in the comments. However, there are some issues that I've encountered.
> 
> On public side:
> 
> 1. In catalogue, clicking on sorting select throws `Uncaught Error: Syntax error, unrecognized expression: #` (but it seems to work nevertheless)

fixed ✅  

> In Admin:
> 
> 1. Clicking on 'Carousel' (http://127.0.0.1:8000/slide) throws an error: `Undefined class constant 'ARTWORKS_DIR' (View: C:\Users\eronisko\workspace\sng\webumenia\resources\views\slides\index.blade.php)`

fixed ✅  

> 2. http://127.0.0.1:8000/article/create generates a bunch of `GET http://127.0.0.1:8000/css/content-slick.css net::ERR_ABORTED 404 (Not Found)`

fixed by https://github.com/SlovakNationalGallery/webumenia.sk/pull/453/commits/9caa89b0d41e83ea48e2b61919ffc327bfa468e7
still works nicely imho

> 3. ~Item editing (e.g. http://127.0.0.1:8000/item/SVK:TMP.1/edit) looks broken -- see JS console~ Fixed in #458
> 4. Every CKEditor form now uses the new (bigger) article font. That's probably not desired.

good point. to be honest - I don't know how to fix it easily. I think it points out to the fact, that also on the frontend it should be displayed unified.... same style for artwork detail description as for authority bio or article content imho
I would prefer to deal with this later - if you don't consider it as a blocker...

eronisko commented on resources/views/dielo.blade.php:
```diff
@@ -3,7 +3,7 @@
 @section('og')
 <meta property="og:title" content="{!! $item->getTitleWithAuthors() !!}" />
 <meta property="og:description"
-    content="{!! $item->work_type; !!}, {{ trans('dielo.item_attr_dating') }}: {!! $item->dating !!}, {{ trans('dielo.item_attr_measurements') }}: {!!  implode(' x ', $item->measurements) !!}" />
```
Oh boy, this entire file needs some code-styling love... (later)

igor-kamil replied:
definitely!

eronisko reviewed (approved):
Looks good, I've created a JIRA for the CKeditor problem https://jira.sng.sk/browse/WEBUMENIA-1567

eronisko commented:
Also, thanks for making all the fixes & very excited for this to go out there! 🤩
